cpp/src/utilities/type_dispatcher.hpp

```diff
@@ -95,45 +256,59 @@ class NVStrings;
  * lambda must be the same, else there will be a compiler error as you would be
  * trying to return different types from the same function.
  * 
- * NOTE: It is undefined behavior if an unsupported or invalid `gdf_dtype` is
+ * @note It is undefined behavior if an unsupported or invalid `gdf_dtype` is
  * supplied.
+ * @note The function @p f does not need to have operator()<T> defined for all
+ * column element types - just the types with which it is used. However, it
+ * must have at least
```

Comment:
"must have at least..."

Reply:
Also, definitely need some documentation on what happens if a `operator()<T>` is invoked for a `T` where the operator is undefined. Is it a compile error? Runtime error?

Reply:
I'm sure I described that somewhere, but I'll describe it here as well.

Reply:
I believe that a runtime error via `release_assert` in device code and throwing an excpetion in host code is the prudent thing to do for now.

Reply:
I wonder if we shouldn't have some wrapper function which does this, instead of having to `#ifdef` everywhere we need it.

Reply:
Well, I'm not going to mess with `CUDA_FAIL()` in this PR. Perhaps some other time.